### ISSUE TITLE
[Select] The macOSVoiceOver doesn't read current value correctly on Safari

### ISSUE THREAD

**Issue by markmagennis** [mui/base-ui#82]
### Steps to reproduce

https://mui.com/base-ui/react-select/

Steps:
1. On MacOS run VoiceOver.
2. Open https://mui.com/base-ui/react-select/ in Safari.
3. Tab to the demo control in the Introduction section.

### Current behavior

When VoiceOver reads the Select control (without it being opened), it announces its value with the last character missing and with a comma added to the beginning. For example for the demo on the MUI site, it announces the three values like so:

- Documentation:   ", Documentatio"
- Components:   ", Component" 
- Features:   ", Feature"

<img width="996" alt="VO screenshot" src="https://github.com/mui/material-ui/assets/25799098/17c26db2-24aa-434a-8ac0-6c2e3899e397">

We have used it in an implementation (unfortunately I can't provide access) and it is sometimes misses out the last two characters and sometimes adds "misspelt" before the value.

This only happens in Safari, not when using Chrome.

### Expected behavior

VoiceOver should announce the value as it is displayed.

### Your environment

MacOS 14.1.1 Sonoma
Safari 17.1

Excuse my ignorance but I don't know how to run `npx @mui/envinfo`. Where should I do this?

**Search keywords**: Select VoiceOver

**Comment by markmagennis** [comment]
@danilo-leal Can I suggest you use "VoiceOver" in the summary rather than changing it to "voice over". This will help users who are searching for VoiceOver issues. Thanks.

**Comment by DiegoAndai** [comment]
Hey, @markmagennis, thanks for the report. I can confirm the behavior, accepting it as a bug.

cc: @michaldudak to prioritize.
